File: Backend/backend-sgac/src/main/java/org/uteq/sgacfinal/entity/TipoEvidencia.java
package org.uteq.sgacfinal.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "tipo_evidencia", schema = "ayudantia")
public class TipoEvidencia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_tipo_evidencia", nullable = false)
    private Integer id;

    @Size(max = 50)
    @NotNull
    @Column(name = "nombre", nullable = false, length = 50)
    private String nombre;

    @Size(max = 10)
    @Column(name = "extension_permitida", length = 10)
    private String extensionPermitida;

    @Column(name = "activo")
    private Boolean activo;

}

File: Frontend/frontend-sgac/src/app/core/models/catalogos/TipoEvidencia.ts
export interface TipoEvidenciaResponse {
  id: number;
  nombre: string;
  extension_permitida: string;
  codigo: string;
  activo: boolean;
}

export interface TipoEvidenciaRequest {
  nombre: string;
  extensionPermitida: string;
  codigo: string;
}
